classes/addressBook.py

```diff
@@ -0,0 +1,260 @@
+        except ValueError:
+            raise ValueError(f'Phone {old_phone} has not been found')
+
+    def find_contact(self, key):
+        # if key in name or key in phone.value or key in str(data.email) or key in data.address:
+        print(f'Result of search on key "{key}" :')
+
+        key_all = False
+
+        for name, data in self.data.items():
+            key_is = False
+            if key in name:
+                key_is = True
+            elif key in str(data.email):
+                key_is = True
+            elif key in data.address:
+                key_is = True
+            elif key in str(data.birthday):
+                key_is = True
+            else:
+                for phone in data.phones:
+                    if key in phone.value:
+                        key_is = True
+            key_all = key_all or key_is
+
+            if key_is:
+                print(f'{self.data[name]}')
+
+        if not key_all:
+
+            print(f'Contacts for {key} not found')
+
+    def how_many_days_to_birthday(self, name):
+
+        record = self.data[name]
+        days_to_birthday = record.days_to_birthday()
+#        print(name,  days_to_birthday)
+        return days_to_birthday
+
+    def remove_address(self, name):
+        try:
+            self.data[name].address = ''
+            print(f'For {name} delete address at AddressBook')
+        except KeyError:
+            raise ValueError(f'Contact {name} has not been found')
+
+    def remove_birthday(self, name):
+        try:
+            self.data[name].birthday = ''
+            print(f'For {name} delete birthday at AddressBook')
+        except KeyError:
+            raise ValueError(f'Contact {name} has not been found')
+
+    def remove_contact(self, name):
+
+        try:
+            self.data.__delitem__(name)
+            print(f'Contact {name} has been deleted')
+
+        except KeyError:
+            raise ValueError(f'Contact {name} has not been found')
+
+    def remove_email(self, name):
+        try:
+            self.data[name].email = ''
+            print(f'For {name} delete email')
+
+        except KeyError:
+            raise ValueError(f'Contact {name} has not been found')
+
+    def remove_phone(self, name, phone):
+
+        try:
+            self.data[name].delete_phone(phone)
+            print(f'For {name} delete phone {phone}')
+
+        except KeyError:
+            raise ValueError(f'Contact {name} has not been found')
+
+    def show_all_contact(self, number_on_page=None):
+        if not number_on_page:
+            number_on_page = 3
+
+        stock = self.iterator()
+        page = 1
+
+        while True:
+            try:
+                print(f"Page:   {page}")
+                for _ in range(number_on_page):
+                    print(next(stock))
+
+                page += 1
+            except StopIteration:
+                print("it's end.No contacts in book")
+                break
+
+    def show_birthdays(self, days):
+
```

Comment:
def show_birthdays(self, days):

    list_birthday = []

    for rec in self.data.values():
        if rec.days_to_birthday() <= days:
            list_birthday.append(rec)

    return list_birthday